### PR TITLE
support click on "hidden lines" text to unfold #186406

### LINKED ISSUE #186406: Let me click on "hidden lines" text to unfold
Testing #186213

I was trying to double-click on the the "123 Hidden Lines" text to unfold that region. I would love to be able to do that if possible

image

### PR DESCRIPTION
**Pull request by Kalmaegi**
Fixes #186406

### CODE REVIEW

**Review by KS-Hoog — APPROVED** [review]

**Review by hediet — DISMISSED** [review]

File: src/vs/editor/browser/widget/diffEditorWidget2/unchangedRanges.ts [hunk 1]
```diff
@@ -182,13 +182,15 @@ class CollapsedCodeOverlayWidget extends ViewZoneOverlayWidget {
...
 			h('div@first', { style: { display: 'flex', justifyContent: 'center', alignItems: 'center' } },
-				[$('a', { title: localize('showAll', 'Show all'), role: 'button', onclick: () => { this._unchangedRegion.showAll(undefined); } }, ...renderLabelWithIcons('$(unfold)'))]
+				[$('a', { title: localize('showAll', 'Show all'), role: 'button', onclick: () => { this.showAll(); } }, ...renderLabelWithIcons('$(unfold)'))]
 			),
 			h('div@others', { style: { display: 'flex', justifyContent: 'center', alignItems: 'center' } }),
 		]),
 		h('div.bottom@bottom', { title: localize('diff.bottom', 'Click or drag to show more below'), role: 'button' }),
 	]);
 
+	private showAll() { this._unchangedRegion.showAll(undefined); }
```
Comment (jrieken) [inline src/vs/editor/browser/widget/diffEditorWidget2/unchangedRanges.ts:192]: yuck - why put methods above constructors

File: src/vs/editor/browser/widget/diffEditorWidget2/unchangedRanges.ts [hunk 2]
```diff
@@ -293,7 +295,15 @@ class CollapsedCodeOverlayWidget extends ViewZoneOverlayWidget {
 			if (!this.hide && true) {
 				const lineCount = _unchangedRegion.getHiddenModifiedRange(reader).length;
 				const linesHiddenText = localize('hiddenLines', '{0} Hidden Lines', lineCount);
-				children.push($('span', { title: linesHiddenText }, linesHiddenText));
+				const span = $('span', { title: localize('diff.hiddenLines.expandAll', 'Double click to show all unchanged region') }, linesHiddenText);
+				addDisposableListener(span, 'dblclick', e => {
```
Comment (jrieken) [inline src/vs/editor/browser/widget/diffEditorWidget2/unchangedRanges.ts:299]: Does this need to be disposed?

**Review by jrieken — COMMENTED** [review]
@hedit left some nit comment

File: src/vs/editor/browser/widget/diffEditorWidget2/unchangedRanges.ts (hunk 1 above)
Comment (weartist) [reply src/vs/editor/browser/widget/diffEditorWidget2/unchangedRanges.ts:192]: Sorry, the previous commit was too hasty, it was indeed my problem

**Review by Tyriar — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**